File: Not Chess [Coding]/chess.py
def encodeIt(input_string):
  keyStr = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/="

  # input_string = escape(input_string)
  output = ""
  chr1 = chr2 = chr3 = ""
  enc1 = enc2 = enc3 = enc4 = ""
  i = 3

  chr1 = ord(input_string[0])
  chr2 = ord(input_string[1])
  chr3 = ord(input_string[2])

  enc1 = chr1 >> 2
  enc2 = ((chr1 & 3) << 4) | (chr2 >> 4)
  enc3 = ((chr2 & 15) << 2) | (chr3 >> 6)
  enc4 = chr3 & 63

  if not str(chr(chr2)).isdigit():
    enc3 = enc4 = 64;
  elif not str(chr(chr3)).isdigit():
    enc4 = 64;

  output += keyStr[enc1] + keyStr[enc2] + keyStr[enc3] + keyStr[enc4]
  chr1 = chr2 = chr3 = ""
  enc1 = enc2 = enc3 = enc4 = ""

  while i < len(input_string):
    chr1 = ord(input_string[i])
    i += 1
    chr2 = ord(input_string[i])
    i += 1
    chr3 = ord(input_string[i])
    i += 1

    enc1 = chr1 >> 2
    enc2 = ((chr1 & 3) << 4) | (chr2 >> 4)
    enc3 = ((chr2 & 15) << 2) | (chr3 >> 6)
    enc4 = chr3 & 63

    if not str(chr(chr2)).isdigit():
      enc3 = enc4 = 64;
    elif not str(chr(chr3)).isdigit():
      enc4 = 64;

    output += keyStr[enc1] + keyStr[enc2] + keyStr[enc3] + keyStr[enc4]
    chr1 = chr2 = chr3 = ""
    enc1 = enc2 = enc3 = enc4 = ""

  return output


def lockpick(match):
  string = [32, 32, 32]
  for ch0 in range(32, 127):
    string[0] = str(chr(ch0))

    for ch1 in range(32, 127):
      string[1] = str(chr(ch1))

      for ch2 in range(32, 127):
        string[2] = str(chr(ch2))

        if encodeIt(''.join(string)) == match:
          print(''.join(string))
          return ''.join(string)

  return "    "


def decodeIt(match):
  seq = 4
  groups = int(len(match) / seq)
  main_string = ""
  left = 0
  right = seq
  for gr in range(groups):
    gr_string = match[left:right]
    main_string += lockpick(gr_string)
    left = right;
    right += seq;

  return main_string;

print(decodeIt("cXVlZW4lMjdzJTIwZ2FtYml0"))

File: Not Chess [Coding]/chess.js
function decodeIt(match) {
  var match_length = match.length / 4;
  var main_string = "";
  for (gr = 0; gr < match_length; gr++) {
    var left = gr * 4;
    var right = left + 4;
    var gr_string = match.substring(left, right);
    var string = "   ";

    loop1:
    for (ch1 = 0; ch1 <= 255; ch1++) {
      ch1_c = String.fromCharCode(ch1);

      loop2:
      for (ch2 = 0; ch2 <= 255; ch2++) {
        ch2_c = String.fromCharCode(ch2);
        
        loop3:
        for (ch3 = 0; ch3 <= 255; ch3++) {
          string = ch1_c + ch2_c + String.fromCharCode(ch3);

          if (encodeIt(string) == gr_string) {
            alert(string)
            break loop1;
          }
        }
      }
    }
    main_string += string;
  }

  return main_string;
}

console.log(decodeIt("cXVlZW4lMjdzJTIwZ2FtYml0"));